src/worker_pool.rs

```diff
@@ -217,6 +256,21 @@ impl WorkerPool {
         // We can start waiting for admission review requests to be evaluated
         let mut next_worker_id = 0;
 
+        if self.policy_evaluation_limit_seconds.is_some() {
```

Comment:
shouldn't this be before doing policy init, and before `verify_policy_settings()` that would already check against an epoch?

Reply:
Good catch, the code you reviewed did NOT have timeouts applied while doing the validation of the policy settings. Unfortunately we cannot move this block of code earlier, because the validation of the settings is done by a single engine. This thread instead goes over all the engines of the workers and increases their internal timer.

I'm going to make a dedicated commit that addresses this issue

Reply:
fixed with 06ba04148125211ce24a0a9d69c3b1b05c566fa7